src/lightning_lite/plugins/environments/slurm_environment.py

```diff
@@ -141,3 +149,36 @@ def resolve_root_node_address(nodes: str) -> str:
         nodes = re.sub(r"\[(.*?)[,-].*\]", "\\1", nodes)  # Take the first node of every node range
         nodes = re.sub(r"\[(.*?)\]", "\\1", nodes)  # handle special case where node range is single number
         return nodes.split(" ")[0].split(",")[0]
+
+    @staticmethod
+    def _validate_srun_used() -> None:
+        """Checks if the `srun` command is available and used.
+
+        Parallel jobs (multi-GPU, multi-node) in SLURM are launched by prepending `srun` in front of the Python command.
+        Not doing so will result in processes hanging, which is a frequent user error. Lightning will emit a warning if
+        `srun` is found but not used.
+        """
+        srun_exists = subprocess.call(["which", "srun"]) == 0
+        srun_used = "SLURM_NTASKS" in os.environ
+        hint = " ".join(["srun", os.path.basename(sys.executable), *sys.argv])[:64]
+        if srun_exists and not srun_used:
+            rank_zero_warn(
+                "The `srun` command is available on your system but is not used. HINT: If your intention is to run"
+                f" Lightning on SLURM, prepend your python command with `srun` like so: {hint} ...",
+                category=PossibleUserWarning,
+            )
+
+    @staticmethod
+    def _validate_srun_variables() -> None:
+        """Checks for conflicting or incorrectly set variables set through `srun` and raises a useful error
+        message.
+
+        Right now, we only check for the most common user errors. See `the srun docs
+        <https://slurm.schedmd.com/srun.html>`_ for a complete list of supported srun variables.
+        """
+        ntasks = int(os.environ.get("SLURM_NTASKS", "1"))
+        if ntasks > 1 and "SLURM_NTASKS_PER_NODE" not in os.environ:
```

Comment:
Why is ntasks okay if `SLURM_NTASKS_PER_NODE` is also set?

The error isn't clear, it says it's not supported but technically it is if both are set? The tests modified below have both set

Reply:
NTASKS is equivalent to WORLD_SIZE and should be there always. The key here is that the user instructs SLURM that it should start a fixed number of processes per node, and not distribute the processes arbitrarily. It is important that Lighting has the same number of processes on each node (the same limitation applies to torchrun/torchelastic). This is what we are requiring here.

In the future this might change (#14078) but currently it is a requirement.